### PR TITLE
GitCommandLog: Hide '-c option="value with space"'

### PR DESCRIPTION
**Pull request by mstv**
## Proposed changes

- GitCommandLog: Detect git configuration options like '-c option="value with space"' and hide them in the list (as other configuration options, too)

### Before

image

### After

image

## Test methodology 

- manual

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/Logging/CommandLog.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ private void LogProcessEnd(int? exitCode = null, Exception? ex = null)
 
     public sealed partial class CommandLogEntry
     {
-        [GeneratedRegex(@"((-c +[^ ]+)|(--no-optional-locks)) *", RegexOptions.ExplicitCapture)]
+        [GeneratedRegex(@"((-c +[^ ""]+(""[^""]*"")?)|(--no-optional-locks)) *", RegexOptions.ExplicitCapture)]
```
Comment (gerhardol) [inline GitCommands/Logging/CommandLog.cs:69]: Sidenote: I fixed the presentation too, in the future PR where this is seen:  https://github.com/gerhardol/gitextensions/blob/tmp/git-coloring/GitUI/Editor/Diff/AnsiEscapeUtilities.cs#L36

We could also require quotes to simplify the regex

I would rather write the regex something like below, will still miss something like '-c "key=\"apa bepa\""' 
```suggestion
        [GeneratedRegex(@"((-c +([^""]+|(""[^""]*"")))|(--no-optional-locks)) *", RegexOptions.ExplicitCapture)]
```

This is just for presentation, so it does not have to be future proof

Comment (mstv) [reply GitCommands/Logging/CommandLog.cs:69]: I had to find (and revert) that "fix" in order to take the screenshots.
I do not like tabs. They cause inconsistent empty space - this time in the lower pane of the Git Command Log.

The proposed regex will not work for quotes inside quotes. I do not want to write a regex for nested strings, which might even need to be escaped.

Comment (RussKie) [reply GitCommands/Logging/CommandLog.cs:69]: Do we have tests which can help us visualise _what_ strings we expect to support and what strings we won't?

Comment (gerhardol) [reply GitCommands/Logging/CommandLog.cs:69]: No, the presentation must be according to what `-c value` allows, which includes spaces
But it is GE that produces these strings, not users

Comment (gerhardol) [reply GitCommands/Logging/CommandLog.cs:69]: Something like the following still fail:
`Default.Add(new GitConfigItem("core.xxx", "\\\"\\\"\""), "diff");`
But it is GE code, no need to make the regex more complicated

**Review by gerhardol — APPROVED** [review]

**Review by gerhardol — APPROVED** [review]

**Pull request merged**